lib/document-downloader.js

```diff
@@ -112,6 +112,9 @@ DocumentDownloader.fetchAndInstall = function (url, dest) {
           extract.end(content);
         });
       }
+      else if (type && type.mime === 'application/gzip') {
+        throw new TypeError('gzip\'ed tarballs not supported!');
```

Comment:
Good point. I fixed that part so now if a type is detected by `file-type` and is not 'application/x-tar', echidna will reject the request.
In case of a manifest and an html file, `file-type` will return a type `null`.